docs/formatter.qmd

```diff
@@ -174,6 +174,298 @@ list(foo, bar)
+But if you're testing this code and you copy and paste it out of the function, then it no longer parses:
+
+``` r
+if (is.null(a))
+  1
+else
+  2
+```
+
+If you try and run this, then you'll see an error like `Error: unexpected 'else'`.
+This is particularly annoying when you're working inside a debugger.
+Most R debuggers allow you to pause inside functions and highlight and run chunks of that function.
+If you're paused inside `fn()` and try to highlight and run the if statement, then it will confusingly fail to parse.
+Autobracing multiline if statements avoids this problem entirely.
+
+## For, while, and repeat loops
+
+Air unconditionally autobraces the body of all R loops.
+
+``` r
+for (i in 1:5) x <- x + i
+
+# Becomes:
+for (i in 1:5) {
+  x <- x + i
+}
+```
+
+``` r
+while (x < 5) x <- x + 1
+
+# Becomes:
+while (x < 5) {
+  x <- x + 1
+}
+```
+
+## Function definitions
+
+Air will autobrace the body of a function definition if:
+
+-   Any existing part of the function definition spans multiple lines
+
+-   The function definition exceeds the line length
+
+``` r
+fn <- function(a, b)
+  a + b
+
+# Becomes:
+fn <- function(a, b) {
+  a + b
+}
+```
+
+``` r
+fn <- function(
+  a,
+  b
+) a + b
+
+# Becomes
+fn <- function(
+  a,
+  b
+) {
+  a + b
+}
+```
+
+``` r
+fn <- function(a_really_long_variable_name, another_really_long_name) a_really_long_variable_name + another_really_long_name
+
+# Becomes:
+fn <- function(
+  a_really_long_variable_name,
+  another_really_long_name
+) {
+  a_really_long_variable_name + another_really_long_name
+}
+```
+
+Short function definitions are allowed on one line and will not be autobraced.
+These are all allowed by Air:
+
+``` r
+add_one <- function(x) x + 1
+
+map_lgl(xs, function(x) is.logical(x) && length(x) == 1L && !is.na(x))
+
+# This includes anonymous functions
+map_lgl(xs, \(x) is.list(x) && length(x) == 0L)
+```
+
+## With persistent line breaks
+
+Autobracing is particularly useful as a code rewriting tool when combined with persistent line breaks.
+Consider:
+
+``` r
+result <- map_lgl(xs, function(x) is.logical(x) && length(x) == 1L && !is.na(x))
+```
+
+This may be easier to read if it spanned across multiple lines.
+You could manually rework this, or you could let Air help you!
+There are two places you could put a persistent line break depending on what your desired final result is:
+
+``` r
+# Adding a line break before `xs` expands the call
+result <- map_lgl(
+  xs, function(x) is.logical(x) && length(x) == 1L && !is.na(x))
+
+# Becomes:
+result <- map_lgl(
+  xs,
+  function(x) is.logical(x) && length(x) == 1L && !is.na(x)
+)
+```
+
+``` r
+# Adding a line break before `is.logical(x)` forces autobracing
+result <- map_lgl(xs, function(x)
+    is.logical(x) && length(x) == 1L && !is.na(x))
+
+# Becomes:
+result <- map_lgl(xs, function(x) {
+  is.logical(x) && length(x) == 1L && !is.na(x)
+})
+```
```

Comment:
I thought this was a cool section

Reply:
yep it's nice you can choose which format fits most for your use case